File: src/net/dinkla/raytracer/math/Point2D.java
package net.dinkla.raytracer.math;

/**
 * Created by IntelliJ IDEA.
 * User: jorndinkla
 * Date: 10.04.2010
 * Time: 15:17:43
 * To change this template use File | Settings | File Templates.
 */
public class Point2D extends Element2D {

    public Point2D(float x, float y) {
        super(x, y);
    }

    public Point2D plus(final Vector2D v) {
        return new Point2D(x + v.x, y + v.y);
    }

    public Point2D minus(final Vector2D v) {
        return new Point2D(x - v.x, y - v.y);
    }

}

File: src/net/dinkla/raytracer/math/Point2D.kt
package net.dinkla.raytracer.math

open class Point2D(x: Double, y: Double) : Element2D(x, y) {

    operator fun unaryMinus(): Point2D {
        return Point2D(-x, -y);
    }

    operator fun plus(v: Vector2D): Point2D {
        return Point2D(x + v.x, y + v.y)
    }

    operator fun minus(v: Vector2D): Point2D {
        return Point2D(x - v.x, y - v.y)
    }

}